doc(1.8.0): add automatic RWX expansion (#1012)
longhorn/website: content/docs/1.8.0/nodes-and-volumes/volumes/expansion.md

#### RWX volume

> As of v1.8.0, Longhorn does support fully automatic online expansion of the filesystem (NFS) for RWX volumes.  However, the feature requires that not only the v1.8.0 Longhorn Manager and CSI components be running, but also the v1.8.0 version of the Share Manager, which manages the NFS export.  If you have upgraded from a previous version, the Share Manager pods (one for each RWX volume) have not been upgraded automatically, to avoid disruption during the upgrade.  

Reviewer: ```suggestion
From v1.8.0, Longhorn supports fully automatic online expansion of the filesystem (NFS) for RWX volumes.  The feature requires:
 
 - v1.8.0 of Longhorn Manager.
 - that CSI components are running.
 - v1.8.0 of the Share Manager, which manages the NFS export.  
 
 If you have upgraded from a previous version, the Share Manager pods (one for each RWX volume) have not been upgraded automatically, to avoid disruption during the upgrade.  
```

Author: Done, with a small tweak.

> After growing the block device, the CSI layer sends a resize command to the Share Manager to grow the filesystem within it.  With a down-rev share-manager, the command is failed with an "unimplemented" error code and no expansion happens.  To get the right image before the expansion, the simplest thing is to force a restart of the pod.  Identify the Share Manager pod of the RWX volume (typically named `share-manager-<volume name>`) and delete it:  

Reviewer: ```suggestion
After growing the block device, the CSI layer sends a resize command to the Share Manager to grow the filesystem within the block device.  With a down-rev share-manager, the command fails with an "unimplemented" error code and so no expansion happens.  To get the right image before the expansion, the simplest thing is to force a restart of the pod.  Identify the Share Manager pod of the RWX volume (typically named `share-manager-<volume name>`) and delete it:  
```

Author: Done.

```shell
kubectl -n longhorn-system delete pod <the share manager pod>
```

> The pod will automatically be recreated using the appropriate version, and the expansion can complete.  Subsequent expansions will not require any further intervention.

Reviewer: ```suggestion
The pod will automatically be recreated using the appropriate version, and the expansion completes.  Further expansions will not require any further intervention.
```

Author: Done.

##### Offline

> It is also still possible to expand the RWX volume offline using these steps:

Reviewer: ```suggestion
It's still possible to expand the RWX volume offline using these steps:
```

Author: Done.

#### RWX volume

From v1.8.0, Longhorn supports fully automatic online expansion of the filesystem (NFS) for RWX volumes.  The feature requires the v1.8.0 versions of these components to be running:

- Longhorn-Manager
- CSI plugin
- Share Manager, which manages the NFS export

If you have upgraded from a previous version, the Share Manager pods (one for each RWX volume) are not upgraded automatically, to avoid disruption during the upgrade. 

After growing the block device, the CSI layer sends a resize command to the Share Manager to grow the filesystem within the block device.  With a down-rev share-manager, the command fails with an "unimplemented" error code and so no expansion happens.  To get the right image before the expansion, the simplest thing is to force a restart of the pod.  Identify the Share Manager pod of the RWX volume (typically named `share-manager-<volume name>`) and delete it:  

```shell
kubectl -n longhorn-system delete pod <the share manager pod>

> ```

Reviewer 2: I know this replies on RWX volume's failover feature. Can we provide two methods
1. Scale down the workloads using the RWX volume
2. Delete the share-manager pod and recreate it. Need to mention why this works.

WDYT?

Author: I think the scale-down is covered by the `Offline` section below.  
Are you asking that we make it clearer that these are alternatives by numbering them 1 and 2?

1. Scale up the workload.

> The reattached volume will have the expanded size.  Further, the Share Manager pod will be recreated with the current version.

Reviewer: ```suggestion
The reattached volume will have the expanded size.  Furthermore, the Share Manager pod will be recreated with the current version.
```

Author: Done.  Thank you.
